### PR TITLE
Remove duplicate doc and clarify JDBC vs REST

### PR DESCRIPTION
**Pull request by tschaub**
## Description

The metastore documentation for the JDBC catalog currently repeats a couple sentences.  In addition, the text refers to the REST metadata catalog when I think it should refer to JDBC.

## Release notes

(X) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by tschaub** [comment]
> Thank you for your pull request and welcome to the Trino community. We require contributors to sign our [Contributor License Agreement](https://github.com/trinodb/cla/raw/master/Trino%20Foundation%20Individual%20CLA.pdf), and we don't seem to have you on file.

I have mailed in a signed individual CLA.

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 1]
```diff
@@ -511,12 +511,8 @@ Refer to [Iceberg repository](https://github.com/apache/iceberg/blob/main/core/s
 for creating those tables.
 :::
 
-At a minimum, `iceberg.jdbc-catalog.driver-class`,
-`iceberg.jdbc-catalog.connection-url`, and
-`iceberg.jdbc-catalog.catalog-name` must be configured. When using any
-database besides PostgreSQL, a JDBC driver jar file must be placed in the plugin
-directory. The following example shows a minimal catalog configuration using an
-Iceberg REST metadata catalog:
```
Comment (tschaub) [inline docs/src/main/sphinx/object-storage/metastores.md:519]: These sentences are repeated in [the paragraph above](https://github.com/trinodb/trino/pull/21379/files#diff-4adfca7ddcbedd7b7d32aeb423f61dc42e61f3fc2d0961b7135bdff2196c3018R498).

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
@cla-bot check

File: docs/src/main/sphinx/object-storage/metastores.md (hunk 1 above)
Comment (tschaub) [reply docs/src/main/sphinx/object-storage/metastores.md:519]: Should `iceberg.jdbc-catalog.default-warehouse-dir` be added to the list of required properties?

Comment (ebyhr) [reply docs/src/main/sphinx/object-storage/metastores.md:519]: Yes, it should be added.

Comment (tschaub) [reply docs/src/main/sphinx/object-storage/metastores.md:519]: Thanks, @ebyhr.  I've added `iceberg.jdbc-catalog.default-warehouse-dir` to the list of required properties in the latest commit.

Is there a separate place for additional documentation about these properties?  I'm struggling to understand how the `iceberg.jdbc-catalog.catalog-name` relates to the name of the catalog derived from the name of the properties file.  For example, in an `example.properties` file, should `iceberg.jdbc-catalog.catalog-name` also be `example`?  I'd be happy to add details clarifying the meaning of these properties if you think this doc is a good place for it.

Comment (ebyhr) [reply docs/src/main/sphinx/object-storage/metastores.md:519]: `iceberg.jdbc-catalog.catalog-name` is unrelated to catalog properties file name in Trino. The property is used for specifying catalog name in Iceberg metadata tables. 
https://github.com/apache/iceberg/blob/ba1cd36fbbd92e35bf0ad1667daa1c18be723521/core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java#L50

**Pull request merged**